### PR TITLE
Changes since last PR: differentiate data-source types by category e.g. LogicalCollection/LogicalView

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/VocBase/LogicalCollection.h [hunk 1]
```diff
@@ -92,6 +92,11 @@ class LogicalCollection: public LogicalDataSource {
  public:
   LogicalCollection() = delete;
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief the category representing a logical collection
+  //////////////////////////////////////////////////////////////////////////////
+  static Category const& category() noexcept;
+
   virtual std::unique_ptr<LogicalCollection> clone() {
     return std::unique_ptr<LogicalCollection>(new LogicalCollection(*this));
```
Comment (gnusi) [inline arangod/VocBase/LogicalCollection.h]: please use `std::make_unique`

File: arangod/VocBase/LogicalDataSource.h [hunk 2]
```diff
@@ -31,13 +31,34 @@ struct TRI_vocbase_t; // forward declaration
 
 namespace arangodb {
 
+namespace velocypack {
+  class StringRef; // forward declaration
```
Comment (gnusi) [inline arangod/VocBase/LogicalDataSource.h]: do not indent forward declaration

File: arangod/RocksDBEngine/RocksDBEngine.cpp [hunk 3]
```diff
@@ -2016,3 +2014,7 @@ void RocksDBEngine::releaseTick(TRI_voc_tick_t tick) {
 }
 
 }  // namespace arangodb
+
+// -----------------------------------------------------------------------------
+// --SECTION--                                                       END-OF-FILE
+// -----------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (danielhlarkin) [reply arangod/RocksDBEngine/RocksDBEngine.cpp]: I have requested in a previous code review that they be removed.

[the remark this answers is not in the document]
Comment (vasiliy-arangodb) [reply arangod/RocksDBEngine/RocksDBEngine.cpp]: > in a previous code review
do you mean the previous pull-request? it was removed in the previous pull-request
investigation showed that this change was added by git when automatic-merging 'devel' into the directory containing the full changeset for #344

File: arangod/VocBase/LogicalDataSource.h (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalDataSource.h]: done

**Comment by vasiliy-arangodb** [comment]
abandoned in favour of https://github.com/arangodb/arangodb/pull/4781 as per review request

**Pull request closed without merging**